Bump pyluwen to 0.8.0, tt-tools-common to 1.5

kaplanz opened the pull request:
Adds support for power-aware luwen, including raising the power state to "high" as needed

> [!CAUTION]
>
> Blocked by tenstorrent/tt-tools-common#61

kaplanz committed:
build: bump required python version to 3.10

Signed-off-by: Zakhary Kaplan <zkaplan@tenstorrent.com>

kaplanz committed:
build(deps): bump pyluwen to 0.8.0, tt-tools-common to 1.5

Signed-off-by: Zakhary Kaplan <zkaplan@tenstorrent.com>

kikkiakki commented on tt_burnin/main.py:
Is the new power management interface incompatible with the GO_BUSY and GO_IDLE arc messages? @jgrowdenTT

kikkiakki replied:
The answer is the new power management interface is replacing GO_BUSY/GO_IDLE. From James:
"We have some compat in FW now for the interaction between those two messages. Need 19.4.0. Prior versions had a compat issue we didn't think about."

So we should probably ensure on our end that we aren't using both at the same time.

jgrowdenTT replied:
[Here's the PR for the compat issue I mentioned. Description of the issue is there. ](https://github.com/tenstorrent/tt-zephyr-platforms/pull/894)

Highly recommend moving to the power ioctl. While it'll work if you use GO_BUSY, you can leave the FW running high AICLK in the case you don't specifically get a chance to clean up with GO_IDLE. If you send the ioctl, and don't get a chance to cleanup, KMD will clean up for you once it sees your handle to the tenstorrent device go away.

kikkiakki commented:
I see some telemetry differences between the old power management (GO_BUSY/GO_IDLE) and new power management (set_power_state) when running tt-burnin on BH.

Old:
<img width="1354" height="212" alt="image" src="https://github.com/user-attachments/assets/a164ccb3-1989-489f-8681-7a09ed0d133f" />

New: AICLK throttles, current is maxed out, voltage is minimized.
<img width="1308" height="204" alt="image" src="https://github.com/user-attachments/assets/ed4f95a9-6f73-4fc5-ad45-fb3a3d227757" />

Is this expected?

(tested on FW v19.4.0)

kikkiakki commented on tt_burnin/main.py:
```diff
@@ -241,6 +245,19 @@ def detect_and_group_devices():
+            sys.exit(1)
+        # Raise power state to high
+        if is_driver_version_at_least(driver, "2.6.0"):
+            device.set_power_state("high")
```
I think I'd move the call to `set_power_state("high")` to where `device.arc_msg(0x52)` is right now in `start_burnin_bh` and have the set power state for driver > 2.6.0, else have the GO_BUSY message.

There is no support for the new power management interface in WH FW, but if we call `set_power_state("high")` on a WH system KMD doesn't throw an error, it just does nothing: https://github.com/tenstorrent/tt-kmd/blob/bda2c96c3a3eb5ee48db2f5a054a5fff83629d49/wormhole.c#L1017

kikkiakki committed:
Only use GO_BUSY/GO_IDLE power messages on BH if tt-kmd < v2.6.0

Enforce the use of the set_power_state interface for kmd >= v2.6.0. Ask users to
update FW if the ioctl fails. It can also fail with EINVAL if the chip doesn't
provide a good response to the message for whatever reason, i.e. it could be
hung, so also mention that a reset might be required.

This will change after we add a fix to the Luwen side to handle this error.

Signed-off-by: Petra Alexson <palexson@tenstorrent.com>

kaplanz commented on tt_burnin/main.py:
```diff
@@ -246,18 +251,21 @@ def detect_and_group_devices():
+    driver = get_driver_version()
+    if is_driver_version_at_least(driver, "2.6.0"):
+        # Raise power state to high (BH)
+        try:
```
Once we update the error handling for `EINVAL` on the `ioctl` we can consider changing the error catch here if necessary.

kaplanz reviewed (commented):
Looks great, thanks for fixing this!
